src/tweenable.rs

```diff
@@ -488,22 +488,17 @@ impl<T> Tweenable<T> for Tween<T> {
         entity: Entity,
         event_writer: &mut EventWriter<TweenCompleted>,
     ) -> TweenState {
-        if !self.is_looping() && self.clock.completed() {
+        if self.clock.state() == TweenState::Completed {
             return TweenState::Completed;
         }
 
         // Tick the animation clock
         let times_completed = self.clock.tick(delta);
-        self.times_completed += times_completed;
-        if times_completed & 1 != 0 && self.tweening_type == TweeningType::PingPong {
+        self.clock.record_completions(times_completed);
```

Comment:
It's a bit awkward that we do `clock.tick()`, get back a times completed, and re-inject that immediately into the clock. Shouldn't that recording be updated during `tick()` instead?

Reply:
Agreed, but I'm just going to delete that code in the next PR so I'd rather not move it.